--- a/Android.bp
+++ b/Android.bp
@@ -121,6 +121,7 @@
         "cryptfs.cpp",
         "CryptoType.cpp",
         "Decrypt.cpp",
+        "nx733j/GatekeeperAidl.cpp",
         "Devmapper.cpp",
         "EncryptInplace.cpp",
         "FileDeviceUtils.cpp",
@@ -197,6 +198,8 @@
         "libkeymint_support",
     ],
     whole_static_libs: [
+        // Carry the generated client into all Make consumers of libvold.a.
+        "android.hardware.gatekeeper-V1-ndk_platform",
         "com.android.sysprop.apex",
         "libc++fs"
     ],
